### PR TITLE
Use full path to zap-x.sh in Docker packaged scans

### PR DESCRIPTION
**Pull request by SOOS-JAlvarez**
This fixes an issue where depending in the shell context / runner that the zap scan is being executed it could not be able to locate the `zap-x.sh` shell script without the parent folder.

How I've troubleshooted the issue:
1 - Have a wrapper image that uses zap as base and modified  https://github.com/zaproxy/zaproxy/blob/main/docker/zap-baseline.py#L428 this line to throw a general error (https://github.com/soos-io/soos-dast)
2 - Have Github Action that uses that wrapper (https://github.com/soos-io/soos-dast-github-action)
3 - Run the github action and see what the issue is 
image
4 - Modify image once again to find proper location of `zap-x.sh`, with `os.system('find / -name zap-x.sh').`, wich turns out to be `/zap/zap-x.sh`
5 - Added /zap/ to  https://github.com/zaproxy/zaproxy/blob/main/docker/zap_common.py#L277
6 - Succesfully able to run the wrapper from the Github Action now and executing the wrapper from everywhere else still works.

This started as a google group thread 
https://groups.google.com/g/zaproxy-users/c/im19A4tPFos

### CODE REVIEW

**Comment by kingthorin** [comment]
To address the DCO requirement you'll need to sign-off the commit(s):

- https://github.com/zaproxy/zaproxy/blob/main/CONTRIBUTING.md#developer-certificate-of-origin
- https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff

**Comment by thc202** [comment]
The changelog should also be updated.

**Comment by psiinon** [comment]
You appear to have pulled in an unrelated change now.

**Comment by SOOS-JAlvarez** [comment]
> You appear to have pulled in an unrelated change now.

Seems that I've pulled that change by mistake when adding the sign off, thanks and sorry for the inconvenience!

**Review by psiinon — APPROVED** [review]

**Comment by SOOS-JAlvarez** [comment]
@psiinon  Hey, sorry to bother, any update about if this being is going to be merged?

**Review by kingthorin — APPROVED** [review]

**Comment by thc202** [comment]
The unrelated changes are still in the branch, better fix up the commits.

File: docker/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,9 @@ All notable changes to the docker containers will be documented in this file.
 ### 2022-11-04
   - Fixed `zap-x.sh` to return the exit code from `zap.sh` instead of `rm -f`
 
+### 2022-11-01
```
Comment (thc202) [inline docker/CHANGELOG.md:7]: This needs to be updated.

Comment (SOOS-JAlvarez) [reply docker/CHANGELOG.md]: > This needs to be updated and moved.

Just updated the changelog and rebased to get rid of the extra commits. thanks!

File: docker/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
 # Changelog
 All notable changes to the docker containers will be documented in this file.
 
+### 2022-11-07
+ - Updated `zap-x.sh` to use full path
```
Comment (thc202) [inline docker/CHANGELOG.md:5]:
```suggestion
 - Updated packaged scans to use full path to `zap-x.sh`.

**Pull request merged**